sw/device/tests/example_test_from_rom.c

```diff
@@ -31,15 +31,18 @@ bool rom_test_main(void) {
   if (kDeviceType != kDeviceSimDV) {
     CHECK_DIF_OK(dif_uart_init(
         mmio_region_from_addr(TOP_EARLGREY_UART0_BASE_ADDR), &uart0));
-    CHECK_DIF_OK(
-        dif_uart_configure(&uart0, (dif_uart_config_t){
-                                       .baudrate = kUartBaudrate,
-                                       .clk_freq_hz = kClockFreqPeripheralHz,
-                                       .parity_enable = kDifToggleDisabled,
-                                       .parity = kDifUartParityEven,
-                                       .tx_enable = kDifToggleEnabled,
-                                       .rx_enable = kDifToggleEnabled,
-                                   }));
+    CHECK(kUartBaudrate <= UINT32_MAX, "kUartBaudrate must fit in uint32_t");
+    CHECK(kClockFreqPeripheralHz <= UINT32_MAX,
+          "kClockFreqPeripheralHz must fit in uint32_t");
```

Comment:
Sadly, these can't be static_assert because the value of `kUartBaudrate` and `kClockFreqPeripheralHz` are `extern` in [sw/device/lib/arch/device.h](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/lib/arch/device.h). The assertion would need to be evaluated at link time.